docs/GUIDE.md

```diff
@@ -0,0 +1,327 @@
+
+## Prerequisites  
+- TEN Agent  
+- Agora Developer account  
+- AWS Developer account  
+- OpenAI Developer account  
+- Understanding of basic Agora RTC  
+- Familiar with SwiftUI programming
+
+## Setup  
+1. Obtain an Azure TTS key from the Microsoft Azure portal -> Azure AI Services -> [Speech Services]. (https://portal.azure.com/#view/Microsoft_Azure_ProjectOxford/CognitiveServicesHub/~/SpeechServices) Create an API key for your region.  
+2. Obtain an OpenAI API key from the [platform dashboard]. (https://platform.openai.com/api-keys)  
+3. Obtain an Agora App ID and certificate from [developer console] (onsole.agora.io/projects).  
+4. Clone the [TEN Agent project](https://github.com/TEN-framework/TEN-Agent) from GitHub.  
+5. Follow the TEN Agent project's instructions to set up the services in your local environment. Use the keys that you obtained from Steps 1-3 to set up the .env file.
+
+You can observe running services from the Docker desktop display, like the following screen shot:  
+![docker](https://github.com/user-attachments/assets/da4525b8-ab3a-4579-b318-591d462b5c9e)
+
+Here the server runs on port 8080 from the ten_agent_dev image. ten_agent_demo and ten_agent_playground are different web frontend applications to use with the server. And the graph designer helps you configure different extension workflows in a visualizing way.  
+  
+## Agent Overview  
+Open Graph Editor on http://localhost:3001, and you should see the following graph of the data flow between the agent modules. The default graph camera.va.openai.azure is selected for illustration.  
+![graph](https://github.com/user-attachments/assets/d10e5db7-c541-42f3-a7b8-0ce7a67afd3a)
+
+The TEN agents take care of all the inter-module communication for you. The agora_rtc module enables your app client's connection to the Agora SD-RTN cloud and transcribes your spoken words into text. The text is sent to OpenAI's module for generative response. Then the response is sent to Azure's Text-to-Speech (TTS) module for artificial human voice so you can hear the response on your device. We will use this graph for our iOS app implementation.
+
+Once you verify that Web version of the AI frontend on your environment, you are ready to create an iOS version for the next step.
+
+## Get Started  
+The completed TEN iOS Demo can be cloned from [this GitHub repo](https://github.com/AgoraIO-Community/TEN-AI-Demo-IOS).  
+Note that the Agora iOS RTC engine is required for the package dependence.  
+![PM](https://github.com/user-attachments/assets/e73fdcb8-9c56-4cb2-9e61-fd8bc0531d29)
+
+Update the config.json file. Essentially, the appId and the serverBaseURL are required.  
```

Comment:
update: 
> Essentially, the appId and the serverBaseURL are required.
to
```
Fill in the `appId` from your Agora account and if you are running the Agent on a different port than the default, make sure to update the `serverBaseURL`.
```

Reply:
What about the channel? there's a key for it in the config, but is it required or is that dynamically generated in the app?

Reply:
I realize there is a mention of this below, but would be good to mention that it will get set later in the code.